internal/services/instance/action_server_action_test.go

```diff
@@ -0,0 +1,126 @@
+package instance_test
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/acctest"
+)
+
+func TestAccActionServer_Basic(t *testing.T) {
+	if acctest.IsRunningOpenTofu() {
+		t.Skip("Skipping TestAccActionServer_Basic because action are not yet supported on OpenTofu")
+	}
+
+	tt := acctest.NewTestTools(t)
+	defer tt.Cleanup()
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProtoV6ProviderFactories: tt.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_instance_server" "main" {
+						name = "test-terraform-datasource-private-nic"
+						type = "DEV1-S"
+						image = "ubuntu_jammy"
+
+					  	lifecycle {
+							action_trigger {
+						  		events  = [after_create]
+						  		actions = [action.scaleway_instance_server_action.main]
+							}
+					  	}
+					}
+
+					action "scaleway_instance_server_action" "main" {
+						config {
+						  	action = "reboot"
+							server_id = scaleway_instance_server.main.id
+						}
+					}`,
+			},
+			{
+				Config: `
+					resource "scaleway_instance_server" "main" {
+						name = "test-terraform-datasource-private-nic"
+						type = "DEV1-S"
+						image = "ubuntu_jammy"
+
+					  	lifecycle {
+							action_trigger {
+						  		events  = [after_create]
+						  		actions = [action.scaleway_instance_server_action.main]
+							}
+					  	}
+					}
+
+					action "scaleway_instance_server_action" "main" {
+						config {
+						  	action = "reboot"
+							server_id = scaleway_instance_server.main.id
+						}
+					}
+
+					data "scaleway_audit_trail_event" "instance" {
+						resource_type = "instance_server"
+						resource_id = scaleway_instance_server.main.id
+						method_name = "ServerAction"
+					}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.scaleway_audit_trail_event.instance", "events.#"),
+					func(state *terraform.State) error {
+						rs, ok := state.RootModule().Resources["data.scaleway_audit_trail_event.instance"]
+						if !ok {
+							return errors.New("not found: data.scaleway_audit_trail_event.instance")
+						}
+
+						countStr := rs.Primary.Attributes["events.#"]
+
+						count, err := strconv.Atoi(countStr)
+						if err != nil {
+							return fmt.Errorf("could not parse events.# as integer: %w", err)
+						}
+
+						if count < 1 {
+							return fmt.Errorf("expected events count > 1, got %d", count)
+						}
+
+						return nil
+					},
+				),
```

Comment:
- Shouldn't the check be `if count < 2`? At this point we expect a `poweron` followed by a `reboot` (and 2 matching events).
- Add check on the latest event's `request_body` matching `"action": "reboot"`
- Maybe add a PreConfig wait to this test (like [this one](https://github.com/scaleway/terraform-provider-scaleway/blob/1d2fc00402f86f84df3b2547773348ffacd10c71/internal/services/audittrail/event_data_source_test.go#L44)) to wait for events to cleanly pop up and secure the test results?